django/db/models/sql/query.py

```diff
@@ -2415,13 +2417,28 @@ def set_annotation_mask(self, names):
         """Set the mask of annotations that will be returned by the SELECT."""
         if names is None:
             self.annotation_select_mask = None
+            if self.selected:
+                # Augment selected with all annotations as the mask was removed.
+                for name in self.annotations:
+                    self.selected[name] = name
```

Comment:
I can also believe I can revert this without a test failure

Reply:
Interesting I'll try to see what this might be needed for by running the full suite.

Reply:
There is effectively is no way to trigger it anymore (this PR is a few months old at this point 😅).

I checked all the `set_annotation_mask` call and the only one that can still pass `names = None` is the one in `get_aggregation` assuming `self.annotation_select_mask = None` prior

https://github.com/django/django/blob/e56a32b89bb7fadffdfaa2cdf12b4863ccd5af9b/django/db/models/sql/query.py#L456-L482

I'm happy to remove the code if you'd like as it's not activated by any code path but I wonder if a good preventive measure here might be drop support for passing `None` in the first place (and adjust `get_aggregation` accordingly) as if a future use case come up we'll definitely run into issues with `Query.selected` being out of sync.

The alternative would be write low level tests for `Query.set_annotation_mask` directly to cover for a potential future use case. Whichever you prefer works for me, just lmk @sarahboyce.

Reply:
Alright I figured I'd just drop the new logic after all as the gymnastic required in `get_aggregation` where getting out of hand.